### PR TITLE
HDDS-10541. Replace GSON with Jackson in debug commands

### PR DESCRIPTION
**Pull request by ArafatKhan2198**
## What changes were proposed in this pull request?
This update addresses compatibility concerns with Java 17 by transitioning from GSON to Jackson in the `Debug-Shell` module. This change simplifies dependencies and ensures smooth operation across different Java versions.

Parent Jira Issue: [HDDS-10538](https://issues.apache.org/jira/browse/HDDS-10538)

This patch streamlines our Debug-Shell Implementation.
Affected Classes:

```
- ChunkKeyHandler.java
- ReadReplicas.java
- ContainerChunkInfo.java
```

## What is the link to the Apache JIRA
https://issues.apache.org/jira/browse/HDDS-10541
## How was this patch tested?
To test out the changes we have to run :- `TestOzoneDebugShell` Integration test.
The forked CI also ran successfully.

### CODE REVIEW

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java [hunk 1]
```diff
@@ -213,15 +209,16 @@ private void downloadReplicasAndCreateManifest(
...
 
-      IOUtils.close(LOG, blockReplicasWithoutChecksum.values());
+      blockReplicasWithoutChecksum.values()
+          .forEach(each -> IOUtils.close(LOG, each));
     }
   }
 
   private Map<DatanodeDetails, OzoneInputStream> replicasOf(BlockID blockID,
-      Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>> replicas) {
+                Map<OmKeyLocationInfo, Map<DatanodeDetails, OzoneInputStream>> replicas) {
```
Comment (tanvipenumudy) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java:221]: Minor nit: I think we can stick with the original indentation, thanks.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java [hunk 2]
```diff
@@ -80,13 +79,13 @@ protected void execute(OzoneClient client, OzoneAddress address)
         XceiverClientManager xceiverClientManager = containerOperationClient.getXceiverClientManager()) {
       OzoneManagerProtocol ozoneManagerClient = client.getObjectStore().getClientProxy().getOzoneManagerClient();
       address.ensureKeyAddress();
-      JsonElement element;
-      JsonObject result = new JsonObject();
+      ObjectMapper objectMapper = new ObjectMapper();
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java:82]: Have you considered using `JsonUtil`, as mentioned in https://github.com/apache/ozone/pull/6147#pullrequestreview-1941518871?

**Review by adoroszlai — COMMENTED** [review]
Thanks @ArafatKhan2198 for splitting this from #6147.

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java (hunk 2 above)
Comment (ArafatKhan2198) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java]: Thanks for reminding!
I have implemented the changes!!

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java [hunk 3]
```diff
@@ -213,10 +205,11 @@ private void downloadReplicasAndCreateManifest(
         }
         replicasJson.add(replicaJson);
       }
-      blockJson.add(JSON_PROPERTY_BLOCK_REPLICAS, replicasJson);
+      blockJson.set(JSON_PROPERTY_BLOCK_REPLICAS, replicasJson);
       blocks.add(blockJson);
 
-      IOUtils.close(LOG, blockReplicasWithoutChecksum.values());
+      blockReplicasWithoutChecksum.values()
+          .forEach(each -> IOUtils.close(LOG, each));
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java:212]: This part of the change is unrelated, only due to merge conflict resolution.  Let's keep the original:

```suggestion
      IOUtils.close(LOG, blockReplicasWithoutChecksum.values());
```

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java [hunk 4]
```diff
@@ -102,8 +100,9 @@ protected void execute(OzoneClient client, OzoneAddress address)
       }
       ContainerLayoutVersion containerLayoutVersion = ContainerLayoutVersion
           .getConfiguredVersion(getConf());
-      JsonArray responseArrayList = new JsonArray();
-      for (OmKeyLocationInfo keyLocation : locationInfos) {
+      ArrayNode responseArrayList = JsonUtils.createArrayNode();
+      for (OmKeyLocationInfo keyLocation : keyInfo.getLatestVersionLocations()
+          .getBlocksLatestVersionOnly()) {
```
Comment (adoroszlai) [inline hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java:105]: Also seems to be unrelated changed.  Can we keep using `locationInfos`?

```suggestion
      for (OmKeyLocationInfo keyLocation : locationInfos) {
```

Comment (ArafatKhan2198) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ChunkKeyHandler.java]: Done!
Thanks!

File: hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java (hunk 3 above)
Comment (ArafatKhan2198) [reply hadoop-ozone/tools/src/main/java/org/apache/hadoop/ozone/debug/ReadReplicas.java]: Done!
Thanks!

**Comment by adoroszlai** [comment]
@dombizita @tanvipenumudy please review

**Review by adoroszlai — APPROVED** [review]

**Comment by adoroszlai** [comment]
Thanks @ArafatKhan2198 for the patch, @tanvipenumudy for the review.

Tested this with Java 17, works well.
